### PR TITLE
Log and trap error on folding expand

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/JSUI-2149

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/FoldingTest.ts [hunk 1]
```diff
@@ -217,6 +217,19 @@ export function FoldingTest() {
...
+      it('should trap the error if a query error happens when getting more results', () => {
+        (test.env.searchEndpoint.search as jasmine.Spy).and.returnValue(Promise.reject("That's a bad query"));
+        expect(() => queryData.results.results[0].moreResults()).not.toThrowError();
+      });
+
+      it('should log an error message if a query error happens when getting more results', async done => {
+        (test.env.searchEndpoint.search as jasmine.Spy).and.returnValue(Promise.reject("That's a bad query"));
+        spyOn(test.cmp.logger, 'error');
+        await queryData.results.results[0].moreResults();
+        expect(test.cmp.logger.error).toHaveBeenCalled();
```
Comment (samisayegh) [inline unitTests/ui/FoldingTest.ts]: Small point, but how about using the slightly more strict check `toHaveBeenCalledTimes(1)` here?

File: unitTests/ui/FoldingTest.ts [hunk 2]
```diff
@@ -217,6 +217,19 @@ export function FoldingTest() {
         queryData = null;
       });
 
+      it('should trap the error if a query error happens when getting more results', () => {
```
Comment (samisayegh) [inline unitTests/ui/FoldingTest.ts]: Would it be better to check that the method returns an empty array rather than whether or not it throws? My thinking is if someone were to remove the `return` statement, this test would still pass.

File: src/ui/Folding/Folding.ts [hunk 3]
```diff
@@ -515,6 +515,10 @@ export class Folding extends Component {
       .then((results: IQueryResults) => {
         this.handlePreprocessMoreResults(results);
         return results.results;
+      })
+      .catch(e => {
```
Comment (huppeje) [inline src/ui/Folding/Folding.ts]: It's a really small detail, but if we want to follow the Coveo TSJS guide don't we want wrap the *e* with parenthesis : `(e)` ?

Comment (olamothe) [reply src/ui/Folding/Folding.ts]: It's mostly a personal opinion, but I don't agree with those kind of "written guides somewhere in a wiki". They never work, and are never followed.

If we want to enforce styling guides like this, it needs to be enforced with tooling/linter/actual scripts/bots that check for whatever arbitrary rule we deem important.

File: unitTests/ui/FoldingTest.ts (hunk 2 above)
Comment (olamothe) [reply unitTests/ui/FoldingTest.ts]: Good point, but I'll add a new test for the empty array check, since checking that it does not "pollute" the console needlessly is still a valid test.
